pkg/operation/botanist/etcdcopybackupstask_test.go

```diff
@@ -0,0 +1,237 @@
+	. "github.com/gardener/gardener/pkg/operation/botanist"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/etcdcopybackupstask"
+	mockedcdcopybackupstask "github.com/gardener/gardener/pkg/operation/botanist/component/etcdcopybackupstask/mock"
+	seedpkg "github.com/gardener/gardener/pkg/operation/seed"
+	shootpkg "github.com/gardener/gardener/pkg/operation/shoot"
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	gomegatypes "github.com/onsi/gomega/types"
+)
+
+var _ = Describe("EtcdCopyBackupsTask", func() {
+	var (
+		ctx              context.Context
+		ctrl             *gomock.Controller
+		c                *mockclient.MockClient
+		reader           *mockclient.MockReader
+		kubernetesClient kubernetes.Interface
+
+		botanist    *Botanist
+		namespace   = "shoot--foo--bar"
+		shootName   = "bar"
+		projectName = "foo"
+		seedName    = "seed"
+	)
+
+	BeforeEach(func() {
+		ctx = context.TODO()
+		ctrl = gomock.NewController(GinkgoT())
+		c = mockclient.NewMockClient(ctrl)
+		reader = mockclient.NewMockReader(ctrl)
+		kubernetesClient = fakeclientset.NewClientSetBuilder().
+			WithClient(c).
+			WithAPIReader(reader).
+			Build()
+
+		botanist = &Botanist{Operation: &operation.Operation{}}
+		botanist.K8sSeedClient = kubernetesClient
+		botanist.Seed = &seedpkg.Seed{}
+		botanist.Shoot = &shootpkg.Shoot{
+			SeedNamespace: namespace,
+		}
+		botanist.Seed.SetInfo(&gardencorev1beta1.Seed{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: seedName,
+			},
+			Spec: gardencorev1beta1.SeedSpec{
+				Backup: &gardencorev1beta1.SeedBackup{
+					Provider: "gcp",
+				},
+			},
+		})
+		botanist.Shoot.SetInfo(&gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      shootName,
+				Namespace: projectName,
+			},
+		})
+	})
+
+	Describe("#DefaultEtcdCopyBackupsTask", func() {
+		It("should create a new EtcdCopyBackupsTask deploy waiter", func() {
+			etcdCopyBackupsTask := botanist.DefaultEtcdCopyBackupsTask()
+			Expect(etcdCopyBackupsTask).NotTo(BeNil())
+		})
+
+		It("should create a new EtcdCopyBackupsTask with correct values", func() {
+			validator := &newEtcdCopyBackupsTaskValidator{
+				expectedClient: Equal(c),
+				expectedLogger: BeNil(),
+				expectedValues: Equal(&etcdcopybackupstask.Values{
+					Name:      botanist.Shoot.GetInfo().Name,
+					Namespace: botanist.Shoot.SeedNamespace,
+				}),
+				expectedWaitInterval:       Equal(etcdcopybackupstask.DefaultInterval),
+				expectedWaitSevereTreshold: Equal(etcdcopybackupstask.DefaultSevereThreshold),
+				expectedWaitTimeout:        Equal(etcdcopybackupstask.DefaultTimeout),
+			}
+
+			oldNewEtcdCopyBackupsTask := NewEtcdCopyBackupsTask
```

Comment:
Nit: You can use `defer test.WithVars()`

Reply:
Done.